lib/OpenQA/Utils.pm

```diff
@@ -224,6 +225,26 @@ sub needle_info {
     $needle->{name}      = $name;
     $needle->{distri}    = $distri;
     $needle->{version}   = $version;
+
+    # After change the needle's properties form to hash, support backward compatibility
+    return $needle unless $needle->{properties};
+
+    # If property's type is hash, just push it to array.
+    # If property's type is string, convert it to a hash.
```

Comment:
This comment is not very useful, i can see what it does looking at the code, it's not very complex code. If this block needs a comment, shouldn't it explain why it does what it does instead?

Reply:
Probably better to just write `# Transform string-workaround-properties into HASH-workaround-properties`